### PR TITLE
Snowball flagging of grouped data

### PR DESCRIPTION
**Pull request by mwregan2**
This PR addresses the case of snowballs occurring within a group. When this happens the center of the snowball does not saturate until the next group. This causes the snowball to not be recognized.
The change is to look at the next group to see if it is saturated.

**Checklist**
- [x] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [x] updated relevant tests
- [x] updated relevant documentation
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

File: src/stcal/jump/jump.py [hunk 1]
```diff
@@ -609,25 +607,38 @@ def make_snowballs(gdq, integration, group, jump_ellipses, sat_ellipses,
...
     # of the saturation circle within the enclosing jump rectangle.
     snowballs = []
+    num_groups = gdq.shape[1]
     for jump in jump_ellipses:
-        if near_edge(jump, low_threshold, high_threshold):
+        # center of jump should be saturated
+        jump_center = jump[0]
+        # if center of the jump ellipse is not saturated in this group and is saturated in
+        # the next group add the jump ellipse to the snowball lisr
+        if group < (num_groups - 1) and \
```
Comment (kmacdonald-stsci) [inline src/stcal/jump/jump.py:616]: Use parentheses, instead of '\' to capture multi-line conditional.

File: src/stcal/jump/jump.py
Comment (kmacdonald-stsci) [inline src/stcal/jump/jump.py:1]: A lot of functions in this file are missing docstrings.

File: src/stcal/jump/jump.py [hunk 2]
```diff
@@ -609,25 +607,38 @@ def make_snowballs(gdq, integration, group, jump_ellipses, sat_ellipses,
...
-                            gdq[integration, :, :, :] = \
-                                extend_saturation(gdq[integration, :, :, :],
-                                                  group, [sat], sat_flag,
-                                                  min_sat_radius,
-                                                  expansion=expansion,
-                                                  max_extended_radius=max_extended_radius)
+    # extend the saturated ellipses that are larger than the min_sat_radius
+    gdq[integration, :, :, :] = \
+        extend_saturation(gdq[integration, :, :, :],
+        group, sat_ellipses, sat_flag,
```
Comment (kmacdonald-stsci) [inline src/stcal/jump/jump.py:637]: Indent function parameters to align with first function parameter.

**Review by kmacdonald-stsci — APPROVED** [review]
The `jump.py` file has many functions missing docstring.  This should be corrected, but is not necessary for this PR.

File: src/stcal/jump/jump.py
Comment (mwregan2) [reply src/stcal/jump/jump.py:1]: If every function has to have full doc strings, a large fraction of the lines in the file will be redundant listings of parameters being passed down to lower level functions.

File: src/stcal/jump/jump.py (hunk 1 above)
Comment (mwregan2) [reply src/stcal/jump/jump.py:616]: fixed

**Review by hbushouse — COMMENTED** [review]
Code looks OK to me. Will give final approval once we've done a regtest run.

**Comment by hbushouse** [comment]
regtest started at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/920/

**Comment by hbushouse** [comment]
OK, I guess none of our test datasets in the regression test suite contain any snowballs that fit this description, because I see no changes at all from this PR. But at least it didn't crash or produce unwanted changes, so I guess it's good.

**Review by hbushouse — APPROVED** [review]

**Pull request merged**